transports/websys-websocket/src/lib.rs

```diff
@@ -0,0 +1,327 @@
+// Copyright (C) 2023 Vince Vasta
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+//! Libp2p websocket transports built on [Websys](https://rustwasm.github.io/wasm-bindgen/web-sys/index.html).
+use futures::{future::Ready, io, prelude::*};
+use libp2p_core::{
+    multiaddr::{Multiaddr, Protocol},
+    transport::{ListenerId, TransportError, TransportEvent},
+};
+use parking_lot::Mutex;
+use send_wrapper::SendWrapper;
+use wasm_bindgen::{prelude::*, JsCast};
+use web_sys::{MessageEvent, WebSocket};
+
+use std::{
+    collections::VecDeque,
+    pin::Pin,
+    sync::Arc,
+    task::Poll,
+    task::{Context, Waker},
+};
+
+/// A Websocket transport that can be used in a Wasm client.
+///
+/// ## Example
+///
+/// To create an authenticated transport instance with Noise protocol and Mplex:
```

Comment:
```suggestion
/// To create an authenticated transport instance with Noise protocol and Yamux:
```

Yamux is used below.